File: 多媒体/imagepickerdemo/imagepickerdemo/ViewController.m
//
//  ViewController.m
//  imagepickerdemo
//
//  Created by YZH on 2018/2/16.
//  Copyright © 2018年 YZH. All rights reserved.
//

#import "ViewController.h"
#import <MobileCoreServices/MobileCoreServices.h>
#import <AVFoundation/AVFoundation.h>
#import <AVKit/AVKit.h>

@interface ViewController ()<UIImagePickerControllerDelegate,UINavigationControllerDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *imageview;
@property(strong,nonatomic)UIImagePickerController *imagePickerCon;
- (IBAction)button1:(UIButton *)sender;

@end

@implementation ViewController
//懒加载:重写
-(UIImagePickerController *)imagePickerCon{
    if (!_imagePickerCon) {
        _imagePickerCon = [[UIImagePickerController alloc] init];
        //采集元类型
        _imagePickerCon.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
        //媒体类型
        _imagePickerCon.mediaTypes = [NSArray arrayWithObject:(__bridge NSString *)kUTTypeImage];
       
        //设置代理
        _imagePickerCon.delegate=self;
    }
    return _imagePickerCon;
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)button1:(UIButton *)sender {
    //通过摄像头
    if([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]){
        self.imagePickerCon.sourceType = UIImagePickerControllerSourceTypeCamera;
    }
    else{
        //通过图片库
        self.imagePickerCon.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
       
    }
    //模态
    [self presentViewController:_imagePickerCon animated:YES completion:nil];
}

//完成采集后会掉处理
- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info{
    //获取媒体类型
    NSString *type = info[UIImagePickerControllerMediaType];
    if ([type isEqualToString:(__bridge NSString *)kUTTypeImage]) {
       // UIImage *image1 = info[UIImagePickerControllerEditedImage];
        //获取采集到的图片
        UIImage *image = info[UIImagePickerControllerOriginalImage];
        //显示
        self.imageview.image=image;
        
    }
    [_imagePickerCon dismissViewControllerAnimated:true completion:nil];
}
//取消采集图片处理
- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker{
    NSLog(@"NO");
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: camera/camera/ViewController.swift
//
//  ViewController.swift
//  camera
//
//  Created by YZH on 2017/12/17.
//  Copyright © 2017年 YZH. All rights reserved.
//

import UIKit
import AVKit
import AVFoundation
import MobileCoreServices

class ViewController: UIViewController,UIImagePickerControllerDelegate,
                                        UINavigationControllerDelegate {
    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var takePictureButton: UIButton!
    var avPlayerViewController:AVPlayerViewController!
    var image:UIImage?
    var movieURL:URL?
    var choose:String?
    @IBAction func shootpictureorlibrary(_ sender: UIButton) {
        pickMediaFromSource(sourceType: .camera)
    }
    @IBAction func selectexistingpictureorvideo(_ sender: UIButton) {
        pickMediaFromSource(sourceType: .photoLibrary)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if !UIImagePickerController.isSourceTypeAvailable(UIImagePickerControllerSourceType.camera) {
            takePictureButton.isHidden=true
        }
    }
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        updateDisplay()
    }
    
    func updateDisplay(){
        if let mediaType=choose{
            if mediaType==kUTTypeImage as String{
                imageView.image=image!
                imageView.isHidden=false
                if avPlayerViewController != nil{
                    avPlayerViewController!.view.isHidden=true
                }
            }//else if mediaType==kUTTypeMovie
        }
    }

    func pickMediaFromSource(sourceType:UIImagePickerControllerSourceType){
        let mediaTypes = UIImagePickerController.availableMediaTypes(for: sourceType)!
        if UIImagePickerController.isSourceTypeAvailable(sourceType) && mediaTypes.count>0 {
            let  picker = UIImagePickerController()
            picker.mediaTypes=mediaTypes
            picker.delegate=self
            picker.allowsEditing=true
            picker.sourceType=sourceType
            present(picker,animated: true,completion: nil)
        }
        else{
            let alertCOntroller = UIAlertController(title:"ERROR",message:"Unsupported",preferredStyle:UIAlertControllerStyle.alert)
            let okAction = UIAlertAction(title:"OK",style:UIAlertActionStyle.cancel,handler:nil)
            alertCOntroller.addAction(okAction)
            present(alertCOntroller,animated: true,completion: nil)
    
        }
    }
    
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
        choose=info[UIImagePickerControllerMediaType] as! String
        if let mediaType = choose{
            if mediaType==kUTTypeImage as String{
                image=info[UIImagePickerControllerEditedImage] as! UIImage
            }
            else if mediaType == kUTTypeMovie as String{
                movieURL = info[UIImagePickerControllerMediaType] as! URL
            }
        }
        picker.dismiss(animated: true, completion: nil)
    }
    
    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
        picker.dismiss(animated: true, completion: nil)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

